### PR TITLE
feat(chisel): `rawstack` / `rs` command

### PR DESCRIPTION
**Pull request by clabby**
# Overview

Adds a new `!rawstack` / `!rs` command to chisel that displays the raw value of a variable's stack allocation.

## Motivation

Feature requested by @transmissions11 

## Solution

1. Clone the current session source.
2. Append the following code to the cloned source's `run()` function, where `to_inspect` is the argument supplied to `!rawstack`
```solidity
bytes32 __raw__;
assembly {
    __raw__ := to_inspect
}
```
3. Inspect the `__raw__` variable.

## Usage
Screenshot from 2022-12-28 14-31-32

### CODE REVIEW

**Comment by transmissions11** [comment]
fireee

**Comment by transmissions11** [comment]
how/does this work w/ memory types like strings or packed structs?

**Comment by clabby** [comment]
> how/does this work w/ memory types like strings or packed structs?

For memory types, it'll just give you the memory pointer (which is the corresponding stack value for any memory var).
```solidity
bytes32 __raw__;
assembly {
    // if `to_inspect` is a memory var, it will equal the pointer to the var's data in memory
    __raw__ := to_inspect
}
```
Included that piece in the help menu for anyone unfamiliar w/ how that works :smile:

File: chisel/src/dispatcher.rs
Comment (mattsse) [inline chisel/src/dispatcher.rs:655]: nit:
```suggestion
                DispatchResult::CommandFailed(
                    "Variable must exist within `run()` function.".to_string())
```

**Review by mattsse — APPROVED** [review]
codewise lgtm, pending approval @transmissions11

File: chisel/src/dispatcher.rs
Comment (clabby) [reply chisel/src/dispatcher.rs:655]: There's definitely some inconsistencies across chisel w/ `String::from(...)` / `.to_owned()` / `.to_string()`. Is `to_string()` what you guys normally go w/ ?

Comment (mattsse) [reply chisel/src/dispatcher.rs]: they are all the same, but `to_string` is preferable because it is immediately clear what it does

Comment (clabby) [reply chisel/src/dispatcher.rs]: Got it- changed this one for now, can go through and change others when I get back home :+1:

**Comment by mattsse** [comment]
there are apparently conflicts, but github doesn't display which files are affected -.-

@clabby could you please merge master and resolve conflicts

**Comment by clabby** [comment]
> there are apparently conflicts, but github doesn't display which files are affected -.-
> 
> @clabby could you please merge master and resolve conflicts

Done :+1:

**Pull request merged**
